radio/src/targets/common/arm/stm32/aux_serial_driver.cpp

```diff
@@ -256,3 +249,124 @@ extern "C" void AUX_SERIAL_USART_IRQHandler(void)
 #endif // PCBI6X
 }
 #endif // AUX_SERIAL
+
+/**
+ * AUX2 Serial
+ * reduced implementation to use only for FlySky Gimball
+*/
+#if defined(AUX2_SERIAL)
+DMAFifo<32> aux2SerialRxFifo __DMA (AUX2_SERIAL_DMA_Channel_RX);
+void (*aux2SerialIdleCb)(void);
+
+void aux2SerialSetup(unsigned int baudrate, bool dma, uint16_t lenght = USART_WordLength_8b, uint16_t parity = USART_Parity_No, uint16_t stop = USART_StopBits_1)
+{
+  USART_InitTypeDef USART_InitStructure;
+  GPIO_InitTypeDef GPIO_InitStructure;
+
+  GPIO_PinAFConfig(AUX2_SERIAL_GPIO, AUX2_SERIAL_GPIO_PinSource_TX | AUX2_SERIAL_GPIO_PinSource_RX, AUX2_SERIAL_GPIO_AF);
+  //  GPIO_PinAFConfig(AUX2_SERIAL_GPIO, AUX2_SERIAL_GPIO_PinSource_TX, AUX2_SERIAL_GPIO_AF);
+
+  GPIO_InitStructure.GPIO_Pin = AUX2_SERIAL_GPIO_PIN_TX | AUX2_SERIAL_GPIO_PIN_RX;
+  GPIO_InitStructure.GPIO_Mode = GPIO_Mode_AF;
+  GPIO_InitStructure.GPIO_OType = GPIO_OType_PP;
+  GPIO_InitStructure.GPIO_PuPd = GPIO_PuPd_UP;
+  GPIO_InitStructure.GPIO_Speed = GPIO_Speed_2MHz;
+  GPIO_Init(AUX2_SERIAL_GPIO, &GPIO_InitStructure);
+
+  USART_InitStructure.USART_BaudRate = baudrate;
+  USART_InitStructure.USART_WordLength = lenght;
+  USART_InitStructure.USART_StopBits = stop;
+  USART_InitStructure.USART_Parity = parity;
+  USART_InitStructure.USART_HardwareFlowControl = USART_HardwareFlowControl_None;
+  USART_InitStructure.USART_Mode = USART_Mode_Tx | USART_Mode_Rx;
+  USART_Init(AUX2_SERIAL_USART, &USART_InitStructure);
+
+  // if (dma) {
+    aux2SerialRxFifo.stream = AUX2_SERIAL_DMA_Channel_RX; // workaround, CNDTR reading do not work otherwise
+    DMA_InitTypeDef DMA_InitStructure;
+    auxSerialRxFifo.clear();
```

Comment:
This line should be aux2SerialRxFifo.clear(); not auxSerialRxFifo.clear(); ---- the number "2" is missing